docs/tracing.md

```diff
@@ -0,0 +1,74 @@
+# Tracing
+
+## Introduction
+
+Instrumentation based tracing is implemented with
+[`log`](https://github.com/rust-lang/log) and
+[`log_instrument`](https://github.com/JonathanWoollett-Light/log-instrument),
+outputting a `Trace` level log when entering and exiting every function.
+
+It is disabled by default at compile-time, as such the
+**tracing functionality has no impact on the release binary.**
+
+To enable tracing, remove `"max_level_debug"` from the features
+for `log` under [`../src/vmm/Cargo.toml`](../src/vmm/Cargo.toml):
+
+```diff
+- log = { version = "0.4.17", features = ["std", "serde", "max_level_debug"] }
++ log = { version = "0.4.17", features = ["std", "serde"] }
+```
```

Comment:
Could we do this via a firecracker feature instead? E.g. if "log-tracing" feature of firecracker is disabled, add `max_level_debug`, otherwise omit it

Reply:
This can be done, it would require a feature on `vmm` like
```
[features]
omit_trace_logs = ["log/max_level_debug"]
```
then a feature on `firecracker` like
```
[features]
default = ["omit_trace_logs"]
omit_trace_logs = ["vmm/omit_trace_logs"]
```
so compiling `firecracker` with `cargo build --no-default-features` would enable trace logs.

This is a bit messy but provides a nicer interaction.

It would be simplified a little if/when we move `vmm` into `firecracker`.

Reply:
Sounds good, I'd be in favor of doing that :)

Reply:
Okay, I've implemented this.